### PR TITLE
Crowdin refactor implementation 

### PR DESCRIPTION
**Pull request by bethanyaconnor**
This PR fundamentally changes how we gather *level* content for translation (all other content types should be unchanged). 

Major changes:
- Building off of #28548 which read from the database for DSL content, we are now reading from the DB for all level content, using an allow-list of scripts.
- Instead of putting all level content in files by their string type, we now place them in files by script.
- The keys *in Crowdin* are now the URLs of the levels themselves in order to provide more context.

Note that after distribution, strings are still put into files by string type. This is a pretty minor decision (as long as the file structure in `dashboard/config/locales/` is flat, it doesn't matter) but there are already on the order of 1000 files in there that are often quite large and I couldn't think of a new way of organizing content that helped either issue without making the other significantly worse.

### CODE REVIEW

File: bin/i18n/sync-codeorg-in.rb [hunk 1]
```diff
@@ -67,18 +66,100 @@ def localize_dsl_content
     Script.all.each do |script|
       next unless ScriptConstants.i18n? script.name
```
Comment (bethanyaconnor) [inline bin/i18n/sync-codeorg-in.rb:67]: Should we also check if the script is "stable" here?

Comment (nkiruka) [reply bin/i18n/sync-codeorg-in.rb:67]: @bethanyaconnor What do you mean by "stable" in this context?

Comment (bethanyaconnor) [reply bin/i18n/sync-codeorg-in.rb:67]: I believe that means the script is out of beta. There's a function `is_stable?` and I believe Course A 2018 is "stable" but Course A 2019 is not, for example.

File: bin/i18n/i18n_script_utils.rb [hunk 2]
```diff
@@ -142,3 +144,21 @@ def restore(source, redacted, dest, *plugins)
...
 end
+
+def get_level_url_key(script, level)
+  script_name = script.name
+  script_level = level.script_levels.find_by_script_id(script.id)
+
+  "https://studio.code.org/s/#{script_name}/stage/#{script_level.stage.relative_position}/puzzle/#{script_level.position}"
+end
+
+def get_level_from_url(url)
```
Comment (Hamms) [inline bin/i18n/i18n_script_utils.rb:191]: is there a specific reason not to do this with a regex?

File: bin/i18n/sync-codeorg-in.rb [hunk 3]
```diff
@@ -67,18 +66,100 @@ def localize_dsl_content
...
+          end
+
         end
+      end
 
-        i18n_strings.deep_merge! get_dsl_i18n_strings(level)
+      script_strings.delete_if {|_key, value| value.nil? || value.try(:empty)}
+
+      script_i18n_directory = "../i18n/locales/source/course_content"
+      if script.version_year
```
Comment (Hamms) [inline bin/i18n/sync-codeorg-in.rb]: oh, this requirement might have gotten lost along the way, but this should be something like

```ruby
if script.version_year
  script_i18n_directory = "#{script_i18n_directory}/#{script.version_year}"
elsif ScriptConstants.script_in_category?(:hoc, script)
  script_i18n_directory = "#{script_i18n_directory}/Hour of Code"
else
  script_i18n_directory = "#{script_i18n_directory}/other"
end
```

File: bin/i18n/sync-codeorg-in.rb [hunk 4]
```diff
@@ -67,18 +66,100 @@ def localize_dsl_content
...
+        script_i18n_directory = "#{script_i18n_directory}/#{script.version_year}"
+      end
+      Dir.mkdir script_i18n_directory unless Dir.exist? script_i18n_directory
```
Comment (Hamms) [inline bin/i18n/sync-codeorg-in.rb]: nit:
```suggestion
      FileUtils.mkdir_p script_i18n_directory unless Dir.exist? script_i18n_directory
```

File: bin/i18n/sync-codeorg-in.rb (hunk 1 above)
Comment (Hamms) [reply bin/i18n/sync-codeorg-in.rb:67]: definitely!

**Review by Hamms — APPROVED** [review]
This looks great! Couple nits, but otherwise I feel real good about this change

File: bin/i18n/sync-codeorg-in.rb (hunk 1 above)
Comment (bethanyaconnor) [reply bin/i18n/sync-codeorg-in.rb:67]: Oh hmm actually it seems like some hoc scripts aren't marked "stable"

File: bin/i18n/i18n_script_utils.rb (hunk 2 above)
Comment (bethanyaconnor) [reply bin/i18n/i18n_script_utils.rb:191]: Changed!

File: bin/i18n/sync-codeorg-in.rb (hunk 3 above)
Comment (bethanyaconnor) [reply bin/i18n/sync-codeorg-in.rb]: Done.

**Comment by bethanyaconnor** [comment]
I was having trouble with YAML so I've switched to JSON like the proof of concept in codeorg-testing. PTAL and let me know if there are any concerns!

**Pull request merged**